vms/platformvm/state/camino_deposit_offer.go

```diff
@@ -72,8 +28,8 @@ func (cs *caminoState) GetDepositOffer(offerID ids.ID) (*DepositOffer, error) {
 	return offer, nil
 }
 
-func (cs *caminoState) GetAllDepositOffers() ([]*DepositOffer, error) {
-	offers := make([]*DepositOffer, len(cs.modifiedDepositOffers))
+func (cs *caminoState) GetAllDepositOffers() ([]*deposit.Offer, error) {
+	offers := make([]*deposit.Offer, len(cs.modifiedDepositOffers))
 
 	for _, offer := range cs.modifiedDepositOffers {
```

Comment:
couldn't this all be done in a single line?
return append(cs.modifiedDepositOffers.List(), cs.depositOffers...) ?

Beside this this solution does not allow modified / removed entries 
I suggest directly using map to filter out dups